src/output/scanline_output_file.rs

```diff
@@ -134,10 +159,73 @@ impl<'a> ScanlineOutputFile<'a> {
             let msg = unsafe { CStr::from_ptr(error_out) };
             Err(Error::Generic(msg.to_string_lossy().into_owned()))
         } else {
+            self.scanlines_written = self.header().data_dimensions().1;
```

Comment:
I don't mean "write extra pixel data" so much as "overwrite the file in-place." For example, an editing application could keep a file open and write to it after every operation to reduce the chance of data loss. Of course, you can always just create a new `ScanlineOutputFile`, but I'd rather enforce that only if necessary. If upstream does not permit such a use, is there a good reason *not* to express that by making this consume self?